common/models/src/predicate/domain.rs

```diff
@@ -344,7 +349,11 @@ impl TimeRanges {
                     new_time_ranges.push(intersect_tr);
                 }
             }
-            return Some(Self::new(new_time_ranges));
+            if new_time_ranges.is_empty() {
```

Comment:
Add cases to `test_time_ranges()`, such as:

```rust
assert_eq!(trs.intersect(&TimeRange::new(4, 5)), None);
assert_eq!(trs.intersect(&TimeRange::new(40, 50)), None);
```

Reply:
will add it